modules/exploits/unix/webapp/drupal_sa_core_2019_003.rb

```diff
@@ -0,0 +1,178 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  # At the moment only targeting node's rest endpoint and hal_json format
+  Rank = NormalRanking
+
+  include Msf::Exploit::Remote::HTTP::Drupal
+  # XXX: CmdStager can't handle badchars
+  include Msf::Exploit::PhpEXE
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Drupal SA-CORE-2019-003 RestWS Remote PHP Code Execution',
+      'Description'    => %q{
+        This module exploits a Drupal RESTful Web Services RCE.
+
+        Drupal < 8.5.11, and < 8.6.10 are vulnerable.
+      },
+      'Author'         => [
+        'Samuel Mortenson', # Vulnerability discovery
+        'Charles Fol',      # Proof of concept (Drupal 8.x)
+        'Rotem Reiss',      # Metasploit module
+        'wvu'				# Author of the drupal_drupalgeddon2 Metasploit module which was used as the base for this module
+      ],
+      'References'     => [
+        ['CVE', '2019-6340'],
+        ['URL', 'https://www.drupal.org/sa-core-2019-003'],
+        ['URL', 'https://www.ambionics.io/blog/drupal8-rce']
+      ],
+      'DisclosureDate' => '2019-02-20',
+      'License'        => MSF_LICENSE,
+      'Platform'       => ['php', 'unix', 'linux'],
+      'Arch'           => [ARCH_PHP, ARCH_CMD, ARCH_X86, ARCH_X64],
+      'Privileged'     => false,
+      'Payload'        => {'BadChars' => '&>\''},
+      'Targets'        => [ ['Automatic', {}] ],
+      'DefaultTarget'  => 0, # Automatic (PHP In-Memory)
+      'DefaultOptions' => {'WfsDelay' => 2},
```

Comment:
Wait only two seconds for a shell?

Reply:
I used [drupal_drupalgeddon2](https://github.com/rapid7/metasploit-framework/blob/aa0ba91d92ae9324c1c0014d39d1372ed8f789c6/modules/exploits/unix/webapp/drupal_drupalgeddon2.rb#L125) as reference.
It works great for me (tested locally with a vulnerable webapp), should I increase it? If so - to how much do you recommend?

Reply:
Don't do that. It was a specific hack.

Reply:
And for the record, a module's `WfsDelay` is normally zero, but it's combined with the payload's, which is non-zero.

Reply:
@wvu-r7 I honestly have no clue what this value should be, this is why I took another exploit as reference :) Can you please suggest a better value for that? (Again, this is my first exploit, so my knowledge in Metasploit exploits and its options is ~0)

Reply:
I would leave it alone or not even set it and use it unless you're sure about it.

In `drupal_drupalgeddon2`, setting `WfsDelay` to two seconds will wait two seconds between each exploit attempt, including two seconds after the final attempt, plus two seconds from the payload handler.

If you don't set `WfsDelay` in the module, the only wait will be from the payload handler, which is two seconds by default.